src/Aspire.Hosting/Dcp/DcpHost.cs

```diff
@@ -366,4 +370,120 @@ void LogLines(in ReadOnlySequence<byte> buffer, out SequencePosition position)
             reader.Complete();
         }
     }
+
+    private void TryShowContainerRuntimeNotification(DcpInfo dcpInfo, CancellationToken cancellationToken)
+    {
+        // Check if the interaction service is available (dashboard enabled)
+        if (!_interactionService.IsAvailable)
+        {
+            return;
+        }
+
+        var containerRuntime = _dcpOptions.ContainerRuntime;
+        if (string.IsNullOrEmpty(containerRuntime))
+        {
+            // Default runtime is Docker
+            containerRuntime = "docker";
+        }
+
+        var installed = dcpInfo.Containers?.Installed ?? false;
+        var running = dcpInfo.Containers?.Running ?? false;
+
+        // Only show notification if container runtime is installed but not running
+        // If not installed, that's usually a more fundamental setup issue that would be addressed differently
+        if (installed && !running)
+        {
+            string title = "Container Runtime Unhealthy";
+            var (message, linkUrl) = DcpDependencyCheck.BuildContainerRuntimeUnhealthyMessage(containerRuntime);
+
+            var options = new NotificationInteractionOptions
+            {
+                Intent = MessageIntent.Error,
+                LinkText = linkUrl is not null ? "Learn more" : null,
+                LinkUrl = linkUrl
+            };
+
+            try
+            {
+                // Create a cancellation token source that can be cancelled when runtime becomes healthy
+                var notificationCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _shutdownCts.Token);
+                
+                // Single background task to show notification and poll for health updates
```

Comment:
What about if container runtime is stopped while the app host is running. Should the notification appear then?

Reply:
Nope. We only query it once today no continuously